### PR TITLE
Add basic support for IEDriver to drive an EdgeChromium IE tab.

### PR DESCRIPTION
**Pull request by stanleyhon**
- [X ] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

This change allows IEDriver to attach to an internet explorer tab hosted within the new chromium based Edge browser.

- This change adds two capabilities to the IE extension capability object which define whether to launch chromium-edge (or just run in normal mode), and where to launch it from.

ie.edgechromium : bool
ie.edgepath : string

for example:

{
  "capabilities": {
    "alwaysMatch": {
         "se:ieOptions": {
             "ie.edgechromium": true,
             "ie.edgepath": "C:\\Users\\user\\appdata\\local\\Microsoft\\Edge SXS\\Application\\msedge.exe"
         }
    }
  }
}

- Some internal bookkeeping has been added to various places to know whether we're working with IE in Edge or normal IE.

Known Issues:
- In this preliminary version we don't support opening new tabs or new windows due to a crash when new windows were opened in the Edge IE tab in Browser::NewWindow3. I added a workaround there for Edge.

### CODE REVIEW

File: cpp/iedriver/Browser.cpp [hunk 1]
```diff
@@ -34,12 +34,13 @@
 
 namespace webdriver {
 
-Browser::Browser(IWebBrowser2* browser, HWND hwnd, HWND session_handle) : DocumentHost(hwnd, session_handle) {
+Browser::Browser(IWebBrowser2* browser, HWND hwnd, HWND session_handle, bool isEdgeChrome) : DocumentHost(hwnd, session_handle) {
```
Comment (bwalderman) [inline cpp/iedriver/Browser.cpp]: Should we rename this to isEdgeChromium (similar elsewhere)?

File: cpp/iedriver/BrowserFactory.cpp [hunk 2]
```diff
@@ -411,11 +467,13 @@ bool BrowserFactory::AttachToBrowser(ProcessWindowInfo* process_window_info,
   return attached;
 }
 
+// TODO STAHON: This might need updating for IE
```
Comment (bwalderman) [inline cpp/iedriver/BrowserFactory.cpp]: Remove comment or open an issue.

File: cpp/iedriver/BrowserFactory.cpp [hunk 3]
```diff
@@ -971,12 +1053,14 @@ BOOL CALLBACK BrowserFactory::FindChildWindowForProcess(HWND hwnd, LPARAM arg) {
   } else {
     DWORD process_id = NULL;
     ::GetWindowThreadProcessId(hwnd, &process_id);
-    if (process_window_info->dwProcessId == process_id) {
+    LOG(DEBUG) << "Looking for " << process_window_info->dwProcessId;
+    // stahon, this needs to not happen or look for the right PID for anaheim
+    //if (process_window_info->dwProcessId == process_id) {
```
Comment (bwalderman) [inline cpp/iedriver/BrowserFactory.cpp]: Delete the commented out lines entirely.

File: cpp/iedriver/WebDriverConstants.h [hunk 4]
```diff
@@ -64,6 +64,8 @@
 #define FILE_UPLOAD_DIALOG_TIMEOUT_CAPABILITY "ie.fileUploadDialogTimeout"
 #define USE_LEGACY_FILE_UPLOAD_DIALOG_HANDLING_CAPABILITY "ie.useLegacyFileUploadDialogHandling"
 #define ENABLE_FULL_PAGE_SCREENSHOT_CAPABILITY "ie.enableFullPageScreenshot"
+#define ATTACH_TO_EDGE_CHROME "ie.edgechrome"
```
Comment (bwalderman) [inline cpp/iedriver/WebDriverConstants.h]: Update to say "Chromium" instead of "Chrome".

**Comment by stanleyhon** [comment]
@jimevans Hi Jim, here's the PR into IEDriver we'd like your input on. Thanks!

**Comment by jimevans** [comment]
I’m aware this is here. I’ve been on holiday and am quite ill today. A little more patience would be appreciated.

**Comment by stanleyhon** [comment]
sorry Jim, didn't mean to rush you.

File: cpp/iedriver/Browser.cpp (hunk 1 above)
Comment (stanleyhon) [reply cpp/iedriver/Browser.cpp]: Fixed.

File: cpp/iedriver/BrowserFactory.cpp (hunk 2 above)
Comment (stanleyhon) [reply cpp/iedriver/BrowserFactory.cpp]: Removed.

File: cpp/iedriver/BrowserFactory.cpp (hunk 3 above)
Comment (stanleyhon) [reply cpp/iedriver/BrowserFactory.cpp]: Fixed.

File: cpp/iedriver/WebDriverConstants.h (hunk 4 above)
Comment (stanleyhon) [reply cpp/iedriver/WebDriverConstants.h]: Fixed.

**Comment by jimevans** [comment]
Squashed and merged as 4a50934. Thank you for the contribution!

**Pull request closed without merging**